pr_agent/git_providers/github_provider.py

```diff
@@ -427,7 +427,49 @@ def publish_inline_comments(self, comments: list[dict], disable_fallback: bool =
                 self._publish_inline_comments_fallback_with_verification(comments)
             except Exception as e:
                 get_logger().error(f"Failed to publish inline code comments fallback, error: {e}")
-                raise e
+                raise e    
+    
+    def get_review_thread_comments(self, comment_id: int) -> list[dict]:
+        """
+        Retrieves all comments in the same thread as the given comment.
+        
+        Args:
+            comment_id: Review comment ID
+                
+        Returns:
+            List of comments in the same thread
+        """
+        try:
+            # Fetch all comments with a single API call
+            all_comments = list(self.pr.get_comments())
+            
+            # Find the target comment by ID
+            target_comment = next((c for c in all_comments if c.id == comment_id), None)
+            if not target_comment:
+                return []
+        
+            # First, identify if this is a reply to another comment
+            root_comment_id = target_comment.raw_data.get("in_reply_to_id", target_comment.id)
+            if root_comment_id != target_comment.id:
+                # If this is a reply, find the root comment
+                root_comment = next((c for c in all_comments if c.id == root_comment_id), None)
+                if root_comment:
+                    target_comment = root_comment
+        
+            # Build the thread - include the root comment and all replies to it
+            thread_comments = [
+                c for c in all_comments if 
+                c.id == target_comment.id or c.raw_data.get("in_reply_to_id") == target_comment.id
+            ]
```

Comment:
Some of the code looks redundant. Can you explain it?

```suggestion
            # Get root comment id
            root_comment_id = target_comment.raw_data.get("in_reply_to_id", target_comment.id)
            # Build the thread - include the root comment and all replies to it
            thread_comments = [
    c for c in all_comments if
    c.id == root_comment_id or c.raw_data.get("in_reply_to_id") == root_comment_id
]
```

Reply:
@ofir-frd It's pointless to look for the root comment object, and since in_reply_to_id is the id of the root comment, the unnecessary process is in the existing code. I'll accept your suggestion.